openlcs/libs/unpack.py

```diff
@@ -184,3 +207,56 @@ def unpack_archives(self, src_dir=None):
             errors.extend(
                 self.unpack_archives_using_atool(src_dir, top_dir=True))
         return errors
+
+    def uncompress_tar_file(self, src_file=None, dest_dir=None, f_type='tar'):
+        """
+        Uncompress tar and tar.gz file according to the file extension.
+        """
+        if f_type == 'tar':
+            cmd = f'tar xvf {src_file} -C {dest_dir}'
+            try:
+                subprocess.call(cmd, shell=True)
+            except subprocess.CalledProcessError:
+                error = f'Failed to uncompress file {src_file}'
+                return '', error
```

Comment:
hmm, if this is where the shortcut `run_cli` resides, please update. also I don't see how the returned tuple is being processed, which is confusing to me.

Reply:
Updated.